### PR TITLE
Schedule monthly usage reports for organizations with active deals

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

File: lms/models/organization_usage.py [hunk 1]
```diff
@@ -33,5 +33,5 @@ class OrganizationUsageReport(CreatedUpdatedMixin, Base):
     """Actual report, in JSON format."""
 
     @classmethod
-    def generate_key(cls, organization, tag, report_start, report_end):
+    def generate_key(cls, organization, tag: str, report_start: date, report_end: date):
```
Comment (marcospri) [inline lms/models/organization_usage.py:36]: Annotated the types here

File: lms/services/organization_usage_report.py [hunk 2]
```diff
@@ -66,28 +101,28 @@ def generate_usage_report(
...
 
         return report
 
-    def usage_report(
-        self,
-        organization: Organization,
-        since: datetime,
-        until: datetime,
-    ):
+    def usage_report(self, organization: Organization, since: date, until: date):
```
Comment (marcospri) [inline lms/services/organization_usage_report.py:125]: dates here as well

File: lms/services/organization_usage_report.py [hunk 3]
```diff
@@ -103,7 +138,11 @@ def usage_report(
         # Of those groups, get the ones that do have annotations in the time period
         groups_with_annos = [
             group.authority_provided_id
-            for group in self._h_api.get_groups(groups_from_org, since, until)
+            for group in self._h_api.get_groups(
+                groups_from_org,
+                datetime.combine(since, datetime.min.time()),  # Start of since date
```
Comment (marcospri) [inline lms/services/organization_usage_report.py:150]: We now use `date` everywhere in LMS but the H api uses `datetime`, make the conversion here.

File: lms/migrations/versions/73f0011260e4_usage_report.py [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
+Revises: b512a5cf64ed
 """
 
 import sqlalchemy as sa
 from alembic import op
 from sqlalchemy.dialects import postgresql
 
 revision = "73f0011260e4"
-down_revision = "329313b38de1"
+down_revision = "b512a5cf64ed"
```
Comment (marcospri) [inline lms/migrations/versions/73f0011260e4_usage_report.py:12]: Another migration was merged before I merged this stream of work one.

Updating the parent to keep a linear migration history.

File: lms/services/organization_usage_report.py [hunk 5]
```diff
@@ -33,16 +34,61 @@ class UsageReportRow:
...
         self._h_api = h_api
 
+    def get(
+        self, organization, tag, since: date, until: date
+    ) -> OrganizationUsageReport | None:
+        key = OrganizationUsageReport.generate_key(organization, tag, since, until)
+        return self.get_by_key(key)
+
+    @staticmethod
+    def monthly_report_dates(
```
Comment (marcospri) [inline lms/services/organization_usage_report.py:52]: Doing all the dates math here

This migh involve to:

- Not call `today()` here but pass the value
- Don't assume `monthly` but pass a period as a value

We don't yet need those so I opted for writing this and a unit test with cases covering different edge cases.

**Review by acelaya — APPROVED** [review]
There's a bit of magic around dates, but that's almost always the case.

I haven't tested it, but the code looks good to me.
